Sources/Core/Microsoft.StreamProcessing/Operators/Clip/PartitionedClipJoinPipe.cs

```diff
@@ -347,7 +347,8 @@ private void ProcessPendingEntries()
                             partition.nextRightTime = this.lastRightCTI;
 
                         UpdateTime(partition, Math.Min(this.lastLeftCTI, this.lastRightCTI));
-                        this.cleanKeys.Add(pKey);
+                        if (partition.IsClean())
+                            this.cleanKeys.Add(pKey);
```

Comment:
Nit: let's either put braces on this or put it on a single line (I'd prefer the latter given how short it is)

Reply:
Maybe it's time to enforce some rules here... if I'm not even sure what the convention is how are external contributors supposed to know. IMO we should enforce braces always, seems to be the most prevalent convention.

---
In reply to: [285213620](https://github.com/microsoft/Trill/pull/89#discussion_r285213620) [](ancestors = 285213620)

Reply:
I'm not generally OK with requiring braces, but I'm OK with requiring either braces or a single line.